libs/judicial-system/court-client/src/lib/courtClient.service.ts

```diff
@@ -76,24 +80,48 @@ export class CourtClientService {
       await this.login(clientId)
     }
 
-    try {
-      const res = await request(this.authenticationToken[clientId])
-
-      return stripResult(res)
-    } catch (error) {
-      if (isRetry) {
-        this.logger.error('Error while calling the court service', error)
-
-        throw error
-      }
-
-      this.logger.warn(
-        'Error while calling the court service - attempting relogin',
-        error,
-      )
-
-      return this.wrappedRequest(clientId, request, true)
-    }
+    return request(this.authenticationToken[clientId])
+      .then((res) => stripResult(res))
+      .catch((reason) => {
+        const { statusCode, body } = reason as {
+          statusCode: number
+          body: string
+        }
+
+        if (
+          !isRetry &&
+          statusCode === 400 &&
+          body?.startsWith('authenticationToken is expired')
+        ) {
+          this.logger.warn(
+            'Error while calling the court service - attempting relogin',
+            { reason },
+          )
+
+          return this.wrappedRequest(clientId, request, true)
+        }
+
+        const { status, statusText } = reason as {
```

Comment:
I think this is a duplication of [this code](https://github.com/island-is/island.is/pull/5651/files#diff-30b1389d58ba77128bdb118fdd0fdc6213b60804863fc6b718dee745b6e39857R86). 

On line 86, consider writing 

```
const { statusCode, body, status, statusText } = reason as {
    statusCode: number
    body: string
    status: number
    statusText: string
}
``` 

instead. Also consider chaining the if statements together with `else if` on line 109 instead of if.